src/common/buffer.cc

```diff
@@ -1225,6 +1211,7 @@ static ceph::spinlock debug_lock;
   void buffer::list::rebuild()
   {
     if (_len == 0) {
+      _carriage = &always_empty_bptr;
```

Comment:
nit, i think if `_len == 0`, `_carriage` should always be `&always_empty_bptr`.

Reply:
There could be a case like:
```cpp
bufferlist bl;
bl.reserve(42);
```
Still, keeping `_carriage` intact shouldn't be an issue in such scenario. I guess the assignment is here mostly because of (mechanical) alternations around `_buffers.clear_and_dispose()`.